### PR TITLE
Document recordSelectOptionsQuery() method

### PR DESCRIPTION
**Pull request by intrepidws**
This PR adds documentation for the `Document recordSelectOptionsQuery()` method, which was added in #3588

### CODE REVIEW

File: packages/admin/docs/02-resources/06-relation-managers.md [hunk 1]
```diff
@@ -395,6 +395,16 @@ In this example, `$action->getRecordSelect()` outputs the select field to pick t
...
 Please ensure that any pivot attributes are listed in the `withPivot()` method of the relationship *and* inverse relationship.
 
+### Scoping attachable models
+
+You may want to apply a scope to the models available for attaching in a relation manager:
+
+```php
+use Filament\Tables\Actions\AttachAction;
+
+AttachAction::make()->recordSelectOptionsQuery(fn ($query) => $query->whereBelongsTo(auth()->user())
```
Comment (danharrin) [inline packages/admin/docs/02-resources/06-relation-managers.md:405]: Please add the `recordSelectOptionsQuery()` method on a new line in both examples, and `$query` should be an instance of `Builder`, and that class needs importing too

Comment (intrepidws) [reply packages/admin/docs/02-resources/06-relation-managers.md]: All set Dan.

**Comment by danharrin** [comment]
Maybe it's worth adding examples for AssociateAction too?

**Comment by intrepidws** [comment]
> Maybe it's worth adding examples for AssociateAction too?

Good call, what do you think of the updates?

**Comment by danharrin** [comment]
It doesnt apply to Detach or Dissociate

**Comment by intrepidws** [comment]
Right, of course it doesn't - sorry. That's what I get for doing this late at night. This is fixed.

**Pull request merged**